lib/modules/manager/pixi/extract.ts

```diff
@@ -0,0 +1,56 @@
+import { logger } from '../../../logger';
+import { getSiblingFileName, localPathExists } from '../../../util/fs';
+import { Result } from '../../../util/result';
+import type { PackageFileContent } from '../types';
+import { type PixiConfig, PixiToml, PyprojectToml } from './schema';
+
+function getUserPixiConfig(
+  content: string,
+  packageFile: string,
+): null | PixiConfig {
+  if (packageFile.endsWith('pyproject.toml')) {
+    const { val, err } = Result.parse(content, PyprojectToml).unwrap();
+    if (err) {
+      logger.debug({ packageFile, err }, `error parsing ${packageFile}`);
+      return null;
+    }
+
+    return val.tool?.pixi;
+  }
+
+  if (packageFile.endsWith('pixi.toml')) {
+    const { val, err } = Result.parse(content, PixiToml).unwrap();
+    if (err) {
+      logger.debug({ packageFile, err }, `error parsing ${packageFile}`);
+      return null;
+    }
+
+    return val;
+  }
+
+  /* v8 ignore next 2 */
```

Comment:
add a test instead

Reply:
this is the code path that expected to failed, should I add a error here can check it?

Reply:
Simply test for a file which does not match the the pattern above. 

Users can override the fileMatch and will then run into this.

Reply:
can you explain how? I can think many cases here

For example, `fileMatch: ['pixi.*.toml', 'pyproject.*.toml']`? or a magic toml file pattern? should we try each case? or just still return null?

Reply:
return `null` will do until the need arises as parsing every file is costly. 
If the need arises this can be changed later.

Reply:
I add a fallback to first try to parse config as pyproject.toml then pixi